### PR TITLE
1507158: Provide Host: in http CONNECT header

### PR DESCRIPTION
**Pull request by jirihnidek**
* Bug fix: https://bugzilla.redhat.com/show_bug.cgi?id=1507158
* TODO: verify this bug fix with proxy server requiring this
  header. Waiting for more info from BZ reporter.

### CODE REVIEW

**Comment by jirihnidek** [comment]
[Bug 1507158 - When connecting through a proxy, subscription-manager does not provide Host: in http CONNECT header](https://trello.com/c/zgNCs82V/677-bug-1507158-when-connecting-through-a-proxy-subscription-manager-does-not-provide-host-in-http-connect-header)

File: src/rhsm/connection.py [hunk 1]
```diff
@@ -499,7 +499,10 @@ def _request(self, request_type, method, info=None, headers=None):
 
         if self.proxy_hostname and self.proxy_port:
             log.debug("Using proxy: %s:%s" % (self.proxy_hostname, self.proxy_port))
-            proxy_headers = {'User-Agent': self.user_agent}
+            proxy_headers = {
+                'User-Agent': self.user_agent,
+                'Host': '%s:%s' % (self.host, safe_int(self.ssl_port))
```
Comment (kahowell) [inline src/rhsm/connection.py]: Please normalize host for IPv6 compatibility as in #1725. (See https://github.com/candlepin/subscription-manager/blob/be328f04c089be2eccc508eba58391643ef07bfd/src/rhsm/connection.py#L327-L334)

Comment (jirihnidek) [reply src/rhsm/connection.py]: Sure thing!

**Comment by kahowell** [comment]
@jirihnidek looks good. Please squash.

**Pull request merged**